src/rewrite.ts

```diff
@@ -0,0 +1,54 @@
+export class RewriteRule {
+  find: RegExp;
+  replace: string;
+
+  constructor(find: RegExp, replace: string) {
+    this.find = find;
+    this.replace = replace;
+  }
+
+  apply(value: string): string {
+    return value.replace(this.find, this.replace);
+  }
+}
+
+export class RewriteRules {
+  private rules: RewriteRule[];
+
+  constructor() {
+    this.rules = [];
+  }
+
+  appendRule(rule: RewriteRule): RewriteRules {
+    this.rules.push(rule);
+    return this;
+  }
+
+  apply<T>(value: T): T {
```

Comment:
**Suggestion:** We could exit this early if `this.rules.length === 0`

Reply:
Good suggestion. Done.